### DISCUSSION TITLE
Suggestion: EMQX (mosquitto alternative)

### DISCUSSION THREAD

**Discussion by FredericMa** [tteck/Proxmox#545] [Ideas]
I've seen that the Home Assistant guys created an addon for EMQX which is an alternative to mosquitto.
It seems well maintained and has actually more stars as mosquitto. Also the UI looks really nice and gives you a lot of administration and monitoring options.

Website: https://www.emqx.io/
Download: https://www.emqx.io/downloads?os=Debian
HA Addon: https://github.com/hassio-addons/repository/tree/master/emqx

**Comment by tteck** [comment]
Done!

**Reply by FredericMa** [reply]
Wow, that's fast! Thanks! 😄 
I tried the script but I get this unknown error:
![image](https://user-images.githubusercontent.com/7840263/192087456-eeb11cd2-66e3-413f-b942-d43c484fe6b7.png)
The installation hang for about 1 or 2 minutes before throwing the error.
I'm using AdGuard as DNS and block port 53 except for AdGuard. Don't know if this could have anything to do with it.

**Reply by tteck** [reply]
Try again, I want to use the Apt package, but sometimes it does that. I went back to the deb package so it should work now.

**Reply by FredericMa** [reply]
That worked, thanks!
I noticed that you now hardcoded the latest release. I just did a quick Google to check out if there is an easy way to get the latest release from Github since you can also get the deb over there.
I found this: https://gist.github.com/steinwaywhw/a4cd19cda655b8249d908261a62687f8
I'm not an expert in this so this might not be useful at all.

**Reply by tteck** [reply]
I'd rather use the `apt` package so that an `apt update && apt upgade` would handle the releases, but it's not working right as you discovered. Maybe next release. For now, `.deb` it is.

**Comment by tteck** [comment]
Neat!

![Screenshot 2022-09-23 8 01 10 PM](https://user-images.githubusercontent.com/21342670/192071167-9c1af2ff-fe43-40fe-8a59-719b241ec517.png)
![Screenshot 2022-09-23 7 41 28 PM](https://user-images.githubusercontent.com/21342670/192071249-8771664f-cdc5-4931-bcbe-052284fadb42.png)
![Screenshot 2022-09-23 7 43 18 PM](https://user-images.githubusercontent.com/21342670/192071121-1a258bf9-e8ed-4265-a592-e515a10fe92e.png)

**Reply by FredericMa** [reply]
Pretty cool, right? I've always wanted a nice way to have some kind of MQTT monitoring so it has finally arrived 😄

**Comment by tteck** [comment]
After 8hrs... smooth as butter!

![Screenshot 2022-09-24 8 00 05 AM](https://user-images.githubusercontent.com/21342670/192096978-1c0614df-ba37-4482-9291-9af1287c032a.png)

**Reply by FredericMa** [reply]
Over 6h now, all looking fine! Thanks for the quick and amazing work!
![image](https://user-images.githubusercontent.com/7840263/192117242-b7a048ed-499f-46df-82f9-4a5aa89e1519.png)

**Comment by guarddog13** [comment]
This is awesome. Thank you for this. Shut down my old mqtt...gave this one same ip and MAC addy. Rebooted things attached to mqtt and loving having this amazing GUI.
